smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/incomings/IncomingsTest.java

```diff
@@ -0,0 +1,257 @@
+        MySink sink = container.select(MySink.class).get();
+
+        await().until(() -> bean.list().size() == 6);
+        await().until(() -> sink.list().size() == 6);
+        assertThat(bean.list()).contains("a", "b", "c", "d", "e", "f");
+        assertThat(sink.list()).contains("A", "B", "C", "D", "E", "F");
+    }
+
+    @Test
+    public void testCombiningIncomingsAndMergeWithAProcessor() {
+        addBeanClass(MySink.class);
+        addBeanClass(ProducerOnA.class);
+        addBeanClass(ProcessorUsingMultipleIncomingsAndMerge.class);
+        addBeanClass(ProducerOnB.class);
+        addBeanClass(SecondProducerForB.class);
+
+        initialize();
+        ProcessorUsingMultipleIncomingsAndMerge bean = container.select(ProcessorUsingMultipleIncomingsAndMerge.class).get();
+        MySink sink = container.select(MySink.class).get();
+
+        await().until(() -> bean.list().size() == 8);
+        await().until(() -> sink.list().size() == 8);
+        assertThat(bean.list()).contains("a", "b", "c", "d", "e", "f", "g", "h");
+        assertThat(sink.list()).contains("A", "B", "C", "D", "E", "F", "G", "H");
+    }
+
+    @Test
+    public void testCombiningIncomingsAndMergeWithASubscriber() {
+        addBeanClass(MySink.class);
+        addBeanClass(ProducerOnA.class);
+        addBeanClass(MyBeanUsingMultipleIncomingsAndMerge.class);
+        addBeanClass(ProducerOnB.class);
+        addBeanClass(SecondProducerForB.class);
+
+        initialize();
+        MyBeanUsingMultipleIncomingsAndMerge bean = container.select(MyBeanUsingMultipleIncomingsAndMerge.class).get();
+
+        await().until(() -> bean.list().size() == 8);
+        assertThat(bean.list()).contains("a", "b", "c", "d", "e", "f", "g", "h");
+    }
+
+    @ApplicationScoped
+    public static class ProducerOnA {
+
+        @Outgoing("a")
+        public Publisher<String> produce() {
+            return Flowable.just("a", "b", "c");
+        }
+
+    }
+
+    @ApplicationScoped
+    public static class ProducerOnB {
+
+        @Outgoing("b")
+        public Publisher<String> produce() {
+            return Flowable.just("d", "e", "f");
+        }
+
+    }
+
+    @ApplicationScoped
+    public static class SecondProducerForB {
+
+        @Outgoing("b")
+        public Publisher<String> produce() {
+            return Flowable.just("g", "h");
+        }
+
+    }
+
+    @ApplicationScoped
+    public static class MyBeanUsingMultipleIncomings {
+
+        private List<String> list = new CopyOnWriteArrayList<>();
+
+        @Incoming("a")
+        @Incoming("b")
+        public void consume(String s) {
+            list.add(s);
+        }
+
+        public List<String> list() {
+            return list;
+        }
+
+    }
+
+    @ApplicationScoped
+    public static class MyBeanUsingMultipleIncomingsAndMerge {
+
+        private List<String> list = new CopyOnWriteArrayList<>();
+
+        @Incoming("a")
```

Comment:
Does one of these need to test `@Incomings`?

Reply:
See my previous comment, do we want @Incomings to be a public API?
What's the other MP spec do?